ts/features/bonus/bpd/saga/index.ts

```diff
@@ -34,19 +40,33 @@ import { handleBpdStartOnboardingSaga } from "./orchestration/onboarding/startOn
 export function* watchBonusBpdSaga(bpdBearerToken: string): SagaIterator {
   const bpdBackendClient = BackendBpdClient(bpdApiUrlPrefix, bpdBearerToken);
 
-  // load citizen details
-  yield takeLatest(
-    bpdLoadActivationStatus.request,
-    getCitizen,
-    bpdBackendClient.find
-  );
-
-  // enroll citizen to the bpd
-  yield takeLatest(
-    bpdEnrollUserToProgram.request,
-    putEnrollCitizen,
-    bpdBackendClient.enrollCitizenIO
-  );
+  if (bpdTechnicalIban) {
+    // load citizen details
+    yield takeLatest(
+      bpdLoadActivationStatus.request,
+      getCitizenV2,
+      bpdBackendClient.findV2
+    );
```

Comment:
Could it save some code?
```suggestion
    yield takeLatest(
      bpdLoadActivationStatus.request,
      bpdTechnicalIban ? getCitizenV2 : getCitizen,
      bpdBackendClient.findV2
    );
```

Reply:
Yes, thank you.
Refactored in [e378bd0](https://github.com/pagopa/io-app/pull/2969/commits/e378bd0c6837fb6f56b3c376e48859adbc6e4742)